### PR TITLE
Add logging to DirectoryEntityResource

### PR DESCRIPTION
**Pull request by gmtcrary**
Add logging to DirectoryEntityResource

PR for CO-487.

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy [hunk 1]
```diff
@@ -64,6 +69,7 @@ class DirectoryEntityResource extends Resource {
                 )
                 responseBuilder = ok(resultObject)
             } catch (LdapException ldapException) {
+                logger.error("Ldap Exception thrown when getting by search query", ldapException)
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy:75]: Should probably log all exceptions, not just LdapException, for consistency with the other apis.

Comment (gmtcrary) [reply src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy:75]: https://github.com/osu-mist/web-api-skeleton/pull/43
Once this PR is thru I plan on merging the generic exception mapper in and leaving the current logging as is.

Comment (magical) [reply src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy:75]: Okay

**Comment by gmtcrary** [comment]
In 0eb8d39 some edits on top of the merge were made to conform to the codenarc rules. (Variable naming, line length issues mostly)

[DirectoryApplication.groovy#L57](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryApplication.groovy#L57)

[DirectoryEntityResource.groovy#L55](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy#L55)

[DirectoryEntityResource.groovy#L39](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy#L39)

[DirectoryEntityDAOTest.groovy#L32](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/test/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAOTest.groovy#L32)

[DirectoryEntityDAOTest.groovy#L51](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/test/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAOTest.groovy#L51)

[DirectoryEntityDAO.groovy#L59](https://github.com/osu-mist/directory-api/blob/0eb8d39e5a60b443ab0c3bbaf242c7f4170a7f73/src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAO.groovy#L59)

**Comment by magical** [comment]
That would have been worth mentioning in the commit message.

File: src/main/groovy/edu/oregonstate/mist/api/jsonapi/IOExceptionMapper.groovy [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import edu.oregonstate.mist.api.Error
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.ws.rs.core.Response
+import javax.ws.rs.ext.ExceptionMapper
+import javax.ws.rs.ext.Provider
+
+/**
```
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/jsonapi/IOExceptionMapper.groovy]: Let's remove this default comment. It doesn't add any value that git log doesn't already provide.

**Pull request merged**